### DISCUSSION TITLE
I encountered some problems using the spyOn method.

### DISCUSSION THREAD

**Discussion by EdivenH** [vitest-dev/vitest#6230] [Q&A] [answered]
The testA function and testB function are exported from a single file.
and testA function internally calls the testB function.

I use vitest test testA function, expect testB function to have been called. But the assertion is failing.

this is demo [https://stackblitz.com/edit/vitejs-vite-2uqszv](https://stackblitz.com/edit/vitejs-vite-2uqszv)

please help me~ thanks

**Comment by AriPerkkio** [comment, marked as answer]
See https://vitest.dev/guide/mocking.html#mocking-pitfalls.

Also your reproduction doesn't have Vitest installed or any test cases 🤷

**Reply by EdivenH** [reply]
sorry~ I update demo.
According to the documentation, this scenario cannot be test.   ~~
